docs/source/manual/core.rst

```diff
@@ -26,28 +26,44 @@ Dropwizard consists mostly of glue code to automatically connect and configure t
 Organizing Your Project
 =======================
 
-In general, we recommend you separate your projects into three Maven modules: ``project-api``,
-``project-client``, and ``project-application``.
+If you plan on developing a client library for other developers to access your service, we recommend
+you separate your projects into three Maven modules: ``project-api``, ``project-client``, and
+``project-application``.
 
 ``project-api`` should contain your :ref:`man-core-representations`; ``project-client`` should use
 those classes and an :ref:`HTTP client <man-client>` to implement a full-fledged client for your
 application, and ``project-application`` should provide the actual application implementation, including
 :ref:`man-core-resources`.
 
-Our applications tend to look like this:
+To give a concrete example of this project structure, let's say we wanted to create a Stripe_-like
+API where clients can issue charges and the server would echo the charge back to the client.
+``stripe-api`` project would hold our ``Charge`` object as both the server and client want to work
+with the charge and to promote code reuse, ``Charge`` objects are stored in a shared module.
+``stripe-app`` is the Dropwizard application. ``stripe-client`` abstracts away the raw HTTP
+interactions and deserialization logic. Instead of using a HTTP client, users of ``stripe-client``
+would just pass in a ``Charge`` object to a function and behind the scenes, ``stripe-client`` will
+call the HTTP endpoint. The client library may also take care of connection pooling, and may
+provide a more friendly way of interpreting error messages. Basically, distributing a client library
+for your app will help other developers integrate more quickly with the service.
+
+If you are not planning on distributing a client library for developers, one
+can combine ``project-api`` and ``project-application`` into a single project,
+which tends to look like this:
 
 * ``com.example.myapplication``:
 
-  * ``api``: :ref:`man-core-representations`.
+  * ``api``: :ref:`man-core-representations`. Request and response bodies.
   * ``cli``: :ref:`man-core-commands`
-  * ``client``: :ref:`Client <man-client>` implementation for your application
-  * ``core``: Domain implementation
+  * ``remote``: :ref:`Client <man-client>` code that accesses external HTTP services.
```

Comment:
> I'd think that having `com.stripe.stripe-client.*` and `com.stripe.stripe-app.client` could cause confusion even though the terminology is correct.

My point too, this is why I ended up using `remote` as package in the application module.

But, as you mentioned, the terminology `client` for that package is indeed correct, at least from a technical prospective. It is also generic enough such that the package can (potentially) host _any_ client for services the application depends upon (i.e. not only REST over HTTP).

> To me, core sounds like a module used in other modules, library sounds like something that is included as part of a larger whole, and app is something that is deployed.

I feel, and I may be wrong, that `core` is kind of more suitable (or at least more common) in the context of frameworks rather than applications. While `library` sounds quite generic (maybe too much?).

Naming is a tough business indeed! :sweat: